dockerfiles/Dockerfile.training

```diff
@@ -0,0 +1,193 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# multi-stage arguments (repeat ARG NAME below)
+ARG MOFED_VERSION=4.7-3.2.9.0
+ARG UCX_VERSION=1.8.0
+ARG OPENMPI_VERSION=4.0.4
+ARG CONDA_VERSION=4.7.10
+ARG MKL_VERSION=2020.1
+ARG NUMPY_VERSION=1.18.5
+ARG ONNX_VERSION=1.7.0
+
+# cuda development image for building sources
+FROM nvidia/cuda:10.2-cudnn7-devel-ubuntu18.04 as build
+RUN apt-get -y update
```

Comment:
This is kind of no-op as Docker will not reuse this layer in next apt install.

Ideally we should always do `apt update && apt install -y <package>` to install latest versions using Dockerfile

Reply:
Good point. I made a copy adjacent each apt-get install.